p2panda-core/src/prune.rs

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use serde::{Deserialize, Serialize};
+
+use crate::{validate_backlink, Header, OperationError};
+
+/// Alternative backlink validation method for logs which allow pruning.
+///
+/// When a "prune flag" is set in an operation, an author signals to others that all operations can
+/// be deleted (including payloads) in that log _before_ it.
+///
+/// As soon as a prune flag was set for an operation we don't expect the header of a backlink,
+/// otherwise we go on with validation as usual.
+///
+/// Use this method instead of [`validate_backlink`] if you want to support prunable logs.
+pub fn validate_prunable_backlink<E>(
```

Comment:
With this method being such a thin wrapper around `validate_backlink`, I almost wonder if we need it at all.... I could imagine an if statement handling this case quite nicely. Maybe that would actually result in clearer code and allow for slightly different behaviour around prune points (maybe a user wants to still validate backlinks when present, even on prune points). 

```rust
let prune_flag: PruneFlag = header.extract();
if prune_flag.is_true() {
  // get backlink...
  validate_backlink(past_header, header)
} else {
  Ok(())
}
```

I think the plus points for of having the specialised method is that we get nice docs which explain what the hell pruning is :smile: and this is a big plus.... but it could also appear somewhere else. 

I'm really not sure though, just floating my initial reaction right now.

Reply:
> I think the plus points for of having the specialised method is that we get nice docs which explain what the hell pruning is 😄 and this is a big plus.... but it could also appear somewhere else.

Thats also what think! I don't mind it's so simple, people can ignore it and implement it themselves (as many other things), but it's nice to explain it here and have a clear reference. It's simple code but a complicated concept with big implications.